integration/swarm/swarm.bats

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bats
+# *-*- Mode: sh; sh-basic-offset: 8; indent-tabs-mode: nil -*-*
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Swarm testing : This will start swarm as well as it will create and
+# run swarm replicas using Nginx
+
+# Image for swarm testing
+nginx_image="gabyct/nginx"
+# Name of service to test swarm
+SERVICE_NAME="testswarm"
+# Number of replicas that will be launch
+number_of_replicas=1
+# Timeout in seconds to verify replicas are running
+timeout=10
+# Retry number for the curl
+number_of_retries=5
+# SHIM PATH
+SHIM_PATH="${SHIM_PATH:-/usr/libexec/kata-containers/kata-shim}"
+
+# This function checks if active processes were
+# left behind by kata-runtime.
+
+check_processes() {
+	process=$1
+	pgrep -f "$process"
+	if [ $? -eq 0 ]; then
+		echo "Found unexpected ${process} present"
+		ps -ef | grep $process
+		return 1
+	fi
+}
+
+setup() {
+	interfaces=$(basename -a /sys/class/net/*)
+	swarm_interface_arg=""
+	for i in ${interfaces[@]}; do
+		if [ "$(cat /sys/class/net/${i}/operstate)" == "up" ]; then
+			swarm_interface_arg="--advertise-addr ${i}"
+			break;
+		fi
+	done
+	docker swarm init ${swarm_interface_arg}
+	nginx_command="hostname > /usr/share/nginx/html/hostname; nginx -g \"daemon off;\""
+	docker service create \
+		--name "${SERVICE_NAME}" --replicas $number_of_replicas \
+		--publish 8080:80 "${nginx_image}" sh -c "$nginx_command"
+	running_regex='Running\s+\d+\s(seconds|minutes)\s+ago'
+	for i in $(seq "$timeout") ; do
```

Comment:
you should check the `$i` value inside the loop to see if we have reached the timeout and if so, return 1?